File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        int n = strs.length;

        for(int i = 0; i < n; i++){
            int[] freq = new int[26];
            for(int j = 0; j < strs[i].length(); j++){
                freq[strs[i].charAt(j) - 'a']++;
            }
            String key = Arrays.toString(freq);
            if(!map.containsKey(key)){
                List<String> list = new ArrayList<>();
                list.add(strs[i]);
                map.put(key, list);
            }
            else{
                map.get(key).add(strs[i]);
            }
        }
        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<vector<int>, vector<string>> grped;

        for(string s : strs){
            vector<int> freq(26, 0);
            for(char ch : s){
                freq[ch - 'a']++;
            }
            grped[freq].push_back(s);
        }
        vector<vector<string>> ans;
        for(auto it : grped){
            ans.push_back(it.second);
        }
        return ans;
    }
};